### PR TITLE
New tool for converting the test results to an HTML page.

### PR DESCRIPTION
**Pull request by feldstj**
Updates #416

### CODE REVIEW

File: python/MaterialXTest/tests_to_html.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+try:
+    # Use pip to install Pillow and Image to enable image diffs
+    from PIL import Image
+    from PIL import ImageChops
+    DIFF_ENABLED = True
+except:
+    DIFF_ENABLED = False
+
+def createDiff(image1Path, image2Path, imageDiffPath):
```
Comment (bernardkwok) [inline python/MaterialXTest/tests_to_html.py]: As mentioned this should be a user defined command. It's also not hard to write our own command.

File: python/MaterialXTest/tests_to_html.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            fh.write("<h1>" + subdir + ":</h1><br>\n")
+            for glslFile, oslFile in zip(glslFiles, oslFiles):
+                fullGlslPath = os.path.join(subdir, glslFile)
+                fullOslPath = os.path.join(subdir, oslFile)
+                if glslFile and oslFile and DIFF_ENABLED and CREATE_DIFF:
+                    diffPath = fullGlslPath[0:-8] + "diff.png"
+                    createDiff(fullGlslPath, fullOslPath, diffPath)
+                else:
+                    diffPath = None
+                fh.write("<table>\n")
```
Comment (bernardkwok) [inline python/MaterialXTest/tests_to_html.py]: If it's okay with you can we turn on table borders. At least for me it's easier to see.
Or allow for a style-sheet to be used as an option.

**Comment by bernardkwok** [comment]
You'll either need to fix the codacy errors or add this new file to the ignore list.

File: python/MaterialXTest/tests_to_html.py (hunk 2 above)
Comment (feldstj) [reply python/MaterialXTest/tests_to_html.py]: Done, also made some other stylesheet changes to make it a bit nicer to look at.

File: python/MaterialXTest/tests_to_html.py (hunk 1 above)
Comment (feldstj) [reply python/MaterialXTest/tests_to_html.py]: I agree with this, was going to do that as a separate code review, is that ok or did you want it as part of this one?

Comment (bernardkwok) [reply python/MaterialXTest/tests_to_html.py]: No. That's fine. It's just a comment.

File: python/MaterialXTest/tests_to_html.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            inputdir = arg
+        elif opt in ("-o", "--ofile"):
+            outputfile = arg
+        elif opt in ("-d"):
+            CREATE_DIFF = True
+        elif opt in ("-t"):
+            ENABLE_TIMESTAMPS = True
+
+    fh = open(outputfile,"w+")
+    fh.write("<html>\n")
```
Comment (bernardkwok) [inline python/MaterialXTest/tests_to_html.py]: Thanks for adding the the style sheet here.

**Pull request merged**